pibooth/plugins/printer_plugin.py

```diff
@@ -36,24 +43,19 @@ def state_wait_do(self, cfg, app, events):
                                cfg.getint('PRINTER', 'max_pages'))
                 return
 
-            with timeit("Send final picture to printer"):
-                app.printer.print_file(app.previous_picture_file,
-                                       cfg.getint('PRINTER', 'pictures_per_page'))
-                app.count.printed += 1
-
-            app.count.remaining_duplicates -= 1
+            self.print(cfg, app)
 
     @pibooth.hookimpl
     def state_processing_enter(self, cfg, app):
         app.count.remaining_duplicates = cfg.getint('PRINTER', 'max_duplicates')
 
+    @pibooth.hookimpl
+    def state_processing_do(self, cfg, app):
+        if app.previous_picture_file and app.printer.is_available() and app.count.remaining_duplicates > 0:
+            for i in range(cfg.getint('PRINTER', 'auto_print')):
+                self.print(cfg, app)
```

Comment:
I wouldn't have put the auto print in the processing state, why don't you put is in the `state_print_enter`?

Reply:
The `print` screen can be skipped if delay is = 0 ,  file shall be sent to printer before `print` state

Reply:
Indeed, maybe in `state_processing_exit` hook then? We already have the count increase using this hook for the same reason (print state can be skipped)